### ISSUE TITLE
Sub-surface scatter does not work in 3.1

### ISSUE THREAD

**Issue by mysticfall** [godotengine/godot#25975]
**Godot version:**
`master` / 869887641f3d

**OS/device including version:**
Manjaro Linux 18.2

**Issue description:**
Sub-surface scattering effect used to work (albeit rather primitively) before, but it doesn't seem to work any more in the current master branch.

The parameter had been quite sensitive before (see #18952) so you'd see its effect even when you set the value as low as 0.2. But now, setting it to the maximum value only makes glossy artifacts and does not produce any expected effect.

This is how a material looks with SSS at `0` (still enabled) : 
![image](https://user-images.githubusercontent.com/2367032/52914538-f2582d80-330c-11e9-9995-b86a6bfbbafc.png)

And this is how it looks when it's set to `1.0`:
![image](https://user-images.githubusercontent.com/2367032/52914543-0734c100-330d-11e9-8da5-74c54504d830.png)

**Comment by clayjohn** [comment]
Can you try with beta 5? SSS uses the depth texture which was broken in beta 1 but was fixed a few days ago and is now in beta 5.

**Comment by mysticfall** [comment]
I just tried it again with d9cf780a but it doesn't seem to change the problem.

**Comment by gamergamesz** [comment]
Using gles 3?

**Comment by mysticfall** [comment]
Yes my project setting says it's GLES 3. And I'm running Godot on a Linux desktop with a NVIDIA 970 GTX.

**Comment by gamergamesz** [comment]
It's working on window 10, nvidia 960m

**Comment by aaronfranke** [comment]
When did it work? 3.0? Can you check different 3.1 betas/alphas?

**Comment by mysticfall** [comment]
Is there anything I need to do to see the change? I tested it again after updating the engine to 8b231b96 but cannot see much difference.

**Comment by reduz** [comment]
It seems to work fine for me. Keep in mind that because of a design constraint it only works when there is an environment present. This will be fixed in Godot 4.0.
![image](https://user-images.githubusercontent.com/6265307/53240970-f377ba80-367e-11e9-9ba5-dbc28d7761c2.png)

**Comment by mysticfall** [comment]
@reduz The scene in which the problem occurs on my end has an environment setup. It also has a GI node and I wonder if it could affect the symptom.

Anyway, thanks for the information. I'll try it again with a new project, and also without a GI probe sometime this weekend and let you know the result.

**Comment by mysticfall** [comment]
I just couldn't wait until the weekend :), so I did a quick test.

I created a new scene with minimal setup with an environment which contains a procedural sky. It doesn't have any GI probe and just contains a directional light and a sphere mesh with a minimalistic spatial material with just an albedo texture applied:

![image](https://user-images.githubusercontent.com/2367032/53241752-ffe32e00-36e5-11e9-8955-f6b9acf45fd4.png)

And I set SSS to 1.0 and tried to compare the visual as I switch on and off the effect. And this is when the scene looks like with SSS turned off:

![image](https://user-images.githubusercontent.com/2367032/53241836-433d9c80-36e6-11e9-8dde-fc84ecadfcc3.png)

And this is when SSS is enabled with the maximum value (1.0):

![image](https://user-images.githubusercontent.com/2367032/53241856-53ee1280-36e6-11e9-8a3e-8226a69e6f6e.png)

They look almost identical, but now I can see there is a slight blur effect around the center of sphere. I tried it with a few other textures and got the same result.

No wonder I did not notice the change as the change only becomes apparent at certain area and in a very subtle manner. As I mentioned in the issue description, it used to work rather (probably too much) drastically before, so no wonder I couldn't see the difference with smaller SSS value and a texture with less features.

So, my question now is, is the changed behaviour something expected? I'm trying to use it to make a skin material look more realistic, but I doubt I could achieve the desired effect with so subtle difference.

**Comment by reduz** [comment]
To me it seems to work like it always did, I don´t think anything changed. This effect takes the diffuse part of the image, blurs it according to how much subsurface scatter is there and then addds it back also according to how much subsurface scatter is there (in a pseudo energy conserving approach). 

It does not blur textures but applies the blured image above blended, similar to how glow works, of course with many specifics to the technique, such as not going over discontinuities. If you use it over something with a very contrasting texture, the effect will be less apparent. It also becomes more apparent when there is more contrast between light and shadow, as light bleeds over the shadow and it makes shadow outlines less diffuse. It's mainly used to simulate stuff like skin, marble, milk, etc. a bit more faithfully.

![image](https://user-images.githubusercontent.com/6265307/53261382-9ac21500-36b2-11e9-9489-8a889503ec3e.png)

I will be closing this issue, as I don't think it's broken but in any case if you find anything else that seems strange or out of place feel free to reopen it.

**Comment by mysticfall** [comment]
@reduz But how can I get the effect look so pronounced as shown in your screenshot? As you can see from my comment, the effect looks so subtle in my project that I can hardly tell if anything has changed without switching between 0 and 1 several times.

And I can't even actually crank the value except for testing, because visual artifacts begin to show up around 0.5.

I think it's not working as expected at least on my end, since I can't make the effect any more notable than what I have shown in the above images now, but it used to be much more pronounced when I reported #18952.

Could you suggest anything else that I can try to make my scene look like yours?

**Comment by mysticfall** [comment]
I decided to create a minimal example so I could ask others to test it and post screenshots. So I recreated the same scene in a new project, and now it begins to work as expected.

I didn't even add an environment node this time, but probably it works because of the default environment the editor is using.

I'll compare the two projects and try removing the differences until I find the cause, then I'll let you know the result.

**Comment by mysticfall** [comment]
Found the cause, finally! It was `Rendering/Quality/Subsurface Scattering/Follow Surface` project setting that was enabled on my original project. When I turned off the option, I could clearly see the effect when I raised the SSS value.

Is it normal that enabling the setting makes SSS minimally effective as I've shown with the above screenshots?

By the way, I take back what I said about #18952 being not reproducible any more. It was just another side effect of that option, so now it looks more or less similar to the screenshots I attached in the other issue:
![image](https://user-images.githubusercontent.com/2367032/53263821-d47c3580-371d-11e9-8a49-494784583d03.png)
